core/src/main/java/org/neo4j/gds/core/concurrency/ParallelUtil.java

```diff
@@ -68,6 +68,8 @@
     private static final long DEFAULT_WAIT_TIME_NANOS = 1000;
     private static final long DEFAULT_MAX_NUMBER_OF_RETRIES = (long) 2.5e11; // about 3 days in micros
 
+    private static ForkJoinPool FJ_INSTANCE;
```

Comment:
I like the approach. We actually so a similar thing with our default `ThreadPoolExecutor` in `org.neo4j.gds.core.concurrency.Pools`. I would suggest, we use the opportunity and move the fork join create / get methods into the `Pools` class, too.

Reply:
In `Pools` we do:

```
public static final ExecutorService DEFAULT = createDefaultPool();
```

and refer to it on the procedure level.

I suggest we do the same for ForkJoin Pool, i.e.

```
public static final ForkJoinPool DEFAULT_FJ_POOL = createForkJoinPool();
```

and refer to that one from e.g. `Pregel` and `HugeMergeSort`.

The `ParallelUtil#stream` can create one pool per call and close it (as it does it now).